frontend/src/models/propertyDefinitionsModel.ts

```diff
@@ -1,163 +1,223 @@
 
 // List of property definitions that are calculated on the backend. These
 // are valid properties that do not exist on events.
-const localPropertyDefinitions: PropertyDefinition[] = [
-    {
+const localProperties: PropertyDefinitionStorage = {
+    $session_duration: {
         id: '$session_duration',
         name: '$session_duration',
         description: 'Duration of the session',
         is_numerical: true,
         is_event_property: false,
         property_type: PropertyType.Duration,
     },
-]
-
-const normaliseToArray = (
-    valueToFormat: Exclude<PropertyFilterValue, null>
-): {
-    valueWasReceivedAsArray: boolean
-    arrayOfPropertyValues: (string | number)[]
-} => {
-    if (Array.isArray(valueToFormat)) {
-        return { arrayOfPropertyValues: valueToFormat, valueWasReceivedAsArray: true }
-    } else {
-        return { arrayOfPropertyValues: [valueToFormat], valueWasReceivedAsArray: false }
-    }
 }
 
 export type FormatPropertyValueForDisplayFunction = (
     propertyName?: BreakdownKeyType,
     valueToFormat?: PropertyFilterValue
 ) => string | string[] | null
 
-export const propertyDefinitionsModel = kea<propertyDefinitionsModelType>({
-    path: ['models', 'propertyDefinitionsModel'],
-    actions: () => ({
-        loadPropertyDefinitions: (initial = false) => ({ initial }),
-        updatePropertyDefinition: (property: PropertyDefinition) => ({ property }),
-    }),
-    loaders: ({ values }) => ({
-        propertyStorage: [
-            { results: [], next: null, count: 0 } as PropertyDefinitionStorage,
-            {
-                loadPropertyDefinitions: async ({ initial }, breakpoint) => {
-                    const url = initial
-                        ? 'api/projects/@current/property_definitions/?limit=5000'
-                        : values.propertyStorage.next
-                    if (!url) {
-                        throw new Error('Incorrect call to propertyDefinitionsLogic.loadPropertyDefinitions')
-                    }
-                    const propertyStorage = await api.get(url)
-                    breakpoint()
-                    return {
-                        count: propertyStorage.count,
-                        results: [...values.propertyStorage.results, ...propertyStorage.results],
-                        next: propertyStorage.next,
-                    }
-                },
-            },
-        ],
+/** Update cached property definition metadata */
+export const updatePropertyDefinitions = (
+    propertyDefinitions: PropertyDefinition[] | PropertyDefinitionStorage
+): void => {
+    propertyDefinitionsModel.findMounted()?.actions.updatePropertyDefinitions(propertyDefinitions)
+}
+
+/** Update cached property definition metadata */
+export const updatePropertyDefinition = (propertyDefinitions: PropertyDefinition): void => {
+    propertyDefinitionsModel.findMounted()?.actions.updatePropertyDefinition(propertyDefinitions)
```

Comment:
very nitty: this could call `propertyDefinitionsModel.findMounted()?.actions.updatePropertyDefinitions([propertyDefinition])` and the reader doesn't have to look for the number of instances of the letter S to understand what is happening

Reply:
And then we remove `actions.updatePropertyDefinition` from the model and there's only one way of doing the thing

Reply:
Makes sense. I removed the singular action from everywhere now, so we must always pass an array.